src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/IUpdaterResult.cs

```diff
@@ -5,6 +5,12 @@ namespace Microsoft.DotNet.UpgradeAssistant
 {
     public interface IUpdaterResult
```

Comment:
@sunandabalu should we just change this to a record? I'm not seeing any behavior that would need to be done through an interface?

Reply:
and that would facilitate a lot  :)

Reply:
IUpdaterResult is implemented differently by updaters other than Winforms Updaters. If we make this a record then does that mean we make separate ones for each of the updaters? I guess the ruleid , name and full description is something that might apply to all updaters as we can pull the data into ResultProviders.

Reply:
Changed a few more places so you can see what would change/happen.
We have to check the ruleId values that you want to use.